aprslib/parsing.py

```diff
@@ -243,10 +243,16 @@ def parse(packet):
                     logger.debug("Parsed as normal position report")
                 else:
                     raise ParseError("invalid format")
-
-            # decode comment
-            body, result = _parse_comment(body)
-            parsed.update(result)
+            # check comment for weather information
+            # Page 62 of the spec
+            if parsed['symbol'] == '_':
+               logger.debug("Attempting to parse weather report from comment")
+               body, result = _parse_comment_weather(body)
+               parsed.update(result)
```

Comment:
Hi,

I've checked in those changes.

Regarding your question re one regex..it probably could, so I'll look at
that now.

Cheers
John


On 04/01/16 13:19, Rossen Georgiev wrote:
> In aprslib/parsing.py
> <https://github.com/rossengeorgiev/aprs-python/pull/9#discussion_r48729377>:
> 
>> @@ -243,10 +243,16 @@ def parse(packet):
>>                      logger.debug("Parsed as normal position report")
>>                  else:
>>                      raise ParseError("invalid format")
>> -
>> -            # decode comment
>> -            body, result = _parse_comment(body)
>> -            parsed.update(result)
>> +            # check comment for weather information
>> +            # Page 62 of the spec
>> +            if parsed['symbol'] == '_':
>> +               logger.debug("Attempting to parse weather report from comment")
>> +               body, result = _parse_comment_weather(body)
>> +               parsed.update(result)
> 
> Ok. Let me have look first, since I haven't had a chance to read the
> section in the spec.
> … <#>
> On Mon, Jan 4, 2016 at 1:09 PM, John Ronan <notifications@github.com>
> wrote: In aprslib/parsing.py <#9 (comment)
> <https://github.com/rossengeorgiev/aprs-python/pull/9#discussion_r48728750>>
> : > @@ -243,10 +243,16 @@ def parse(packet): > logger.debug("Parsed as
> normal position report") > else: > raise ParseError("invalid format") >
> - > - # decode comment > - body, result = _parse_comment(body) > -
> parsed.update(result) > + # check comment for weather information > + #
> Page 62 of the spec > + if parsed['symbol'] == '_': > +
> logger.debug("Attempting to parse weather report from comment") > +
> body, result = _parse_comment_weather(body) > + parsed.update(result)
> I'll fix those issues and get back to you. I guess I must not have
> opened them since I installed the pep8 plugin. No, it does not cover
> everything. - It does not cover the RAW weather report format. - It does
> not decode storm data - It does not decode NWS Bulletins Regarding the
> Regex, I have no idea. I had great fun figuring out what is there. I'm
> not sure where to start, but I'm happy to investigate it further. If you
> are happy enough with the niave testing code. I could go through the
> spec and put all the examples into the tests just to be sure. — Reply to
> this email directly or view it on GitHub
> <https://github.com/rossengeorgiev/aprs-python/pull/9/files#r48728750>.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rossengeorgiev/aprs-python/pull/9/files#r48729377>.
>